chains/evm/contracts/offRamp/OffRampOverSuperchainInterop.sol

```diff
@@ -0,0 +1,217 @@
+    }
+    // Validate that the message is meant for this chain.
+    if (message.header.destChainSelector != i_chainSelector) {
+      revert InvalidDestChainSelector(message.header.destChainSelector, i_chainSelector);
+    }
+
+    (Identifier memory identifier, bytes32 logHash) = _constructProofs(message, report.proofs);
+    address onRampAddress = abi.decode(_getEnabledSourceChainConfig(sourceChainSelector).onRamp, (address));
+
+    // Validate that the message was emitted by the corresponding OnRamp.
+    if (identifier.origin != onRampAddress) {
+      revert InvalidSourceOnRamp(sourceChainSelector, identifier.origin);
+    }
+    // Validate that the chainId maps to the expected sourceChainSelector
+    uint256 expectedChainId = s_sourceChainSelectorToChainId[sourceChainSelector];
+    if (expectedChainId == 0) {
+      revert ChainIdNotConfiguredForSelector(sourceChainSelector);
+    }
+    if (expectedChainId != identifier.chainId) {
+      revert ChainIdMismatch(sourceChainSelector, identifier.chainId, expectedChainId);
+    }
+
+    // SECURITY CRITICAL CHECK.
+    // Validate the exact log was emitted on the source chain.
+    i_crossL2Inbox.validateMessage(identifier, logHash);
+
+    hashedLeaves = new bytes32[](1);
+    hashedLeaves[0] = SuperchainInterop._hashInteropMessage(message, onRampAddress);
+
+    // Non-zero timestamp signals the message is verified.
+    // Because there is no Commit timestamp, the timestamp of the message is taken from the time it is sent.
+    // If this OffRamp only accepts low-latency messages from source chains within OP Mesh,
+    // this will be very close to the commit timestamp.
+    // If this OffRamp accepts messages from high-safety-latency sources, e.g alt OP L2s not in OP Mesh,
+    // `permissionLessExecutionThresholdSeconds` needs to be adjusted.
+    return (identifier.timestamp, hashedLeaves);
+  }
+
+  /// @notice Commit is not supported by OffRamp over Superchain Interop, it is replaced by CrossL2Inbox.
+  /// @dev This function is explicitly removed to allow the compiler's UnusedPruner step to remove most
+  /// of the commit-related code from the contract bytecode size.
+  function commit(
+    bytes32[2] calldata,
+    bytes calldata,
+    bytes32[] calldata,
+    bytes32[] calldata,
+    bytes32
+  ) external pure override {
+    revert OperationNotSupportedByThisOffRampType();
+  }
+
+  // ================================================================
+  // │                        Custom Configs                        │
+  // ================================================================
+
+  /// @notice Updates sourceChainSelector to chainId mapping.
+  /// @param chainSelectorsToUnset Array of selectors to remove from the mapping.
+  /// @param chainSelectorsToSet Array of selector to chainId mappings to add.
+  function applyChainSelectorToChainIdConfigUpdates(
+    uint64[] memory chainSelectorsToUnset,
```

Comment:
nit: we never use set/unset wording in any of our contract, probably best to stay consistent